### PR TITLE
Issue #7224 Show POI category icons for Add POI quick action

### PR DESCRIPTION
**Pull request by Dima-1**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java [hunk 1]
```diff
@@ -79,6 +80,29 @@ public AddPOIAction(QuickAction quickAction) {
 		super(quickAction);
 	}
 
+	@Override
+	public int getIconRes(Context context) {
+		if (context instanceof MapActivity) {
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: Why do we need this check?

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java [hunk 2]
```diff
@@ -79,6 +80,29 @@ public AddPOIAction(QuickAction quickAction) {
...
+			final OsmandApplication application = (OsmandApplication) (context).getApplicationContext();
+			final MapPoiTypes poiTypes = application.getPoiTypes();
+			final Map<String, PoiType> allTranslatedNames = poiTypes.getAllTranslatedNames(true);
+			PoiCategory category = getCategory(allTranslatedNames);
+
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: Redundant line break

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java [hunk 3]
```diff
@@ -79,6 +80,29 @@ public AddPOIAction(QuickAction quickAction) {
...
+			if (category != null) {
+
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: Redundant line break

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java [hunk 4]
```diff
@@ -79,6 +80,29 @@ public AddPOIAction(QuickAction quickAction) {
...
+				category.getIconKeyName();
+
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: Redundant line break

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java [hunk 5]
```diff
@@ -79,6 +80,29 @@ public AddPOIAction(QuickAction quickAction) {
...
+	public int getIconRes(Context context) {
+		if (context instanceof MapActivity) {
+			final OsmandApplication application = (OsmandApplication) (context).getApplicationContext();
+			final MapPoiTypes poiTypes = application.getPoiTypes();
+			final Map<String, PoiType> allTranslatedNames = poiTypes.getAllTranslatedNames(true);
+			PoiCategory category = getCategory(allTranslatedNames);
+
+			if (category != null) {
+
+				category.getIconKeyName();
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: ?

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java [hunk 6]
```diff
@@ -79,6 +80,29 @@ public AddPOIAction(QuickAction quickAction) {
...
+
+				String res = category.getIconKeyName();
+				if (res instanceof String && RenderingIcons.containsBigIcon(res)) {
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: what for "res instanceof String" ?

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java [hunk 7]
```diff
@@ -79,6 +80,29 @@ public AddPOIAction(QuickAction quickAction) {
 		super(quickAction);
 	}
 
+	@Override
+	public int getIconRes(Context context) {
+		if (context instanceof MapActivity) {
+			final OsmandApplication application = (OsmandApplication) (context).getApplicationContext();
+			final MapPoiTypes poiTypes = application.getPoiTypes();
+			final Map<String, PoiType> allTranslatedNames = poiTypes.getAllTranslatedNames(true);
```
Comment (max-klaus) [inline OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: I suggest to extract poiTypes  and allTranslatedNames to class field. Do the same in drawUI method.

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java (hunk 1 above)
Comment (Dima-1) [reply OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: Yes it's redundant. Context is always instance of MapActivity.

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java (hunk 5 above)
Comment (Dima-1) [reply OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: getCategory can return null

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java (hunk 6 above)
Comment (Dima-1) [reply OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: It's remains from ShowHidePoiAction) with redundant line breaks. Do i need to check res for null?

File: OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java (hunk 7 above)
Comment (Dima-1) [reply OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java]: Where is the best place to initialize them?

**Pull request merged**